src/wallet/rpcdump.cpp

```diff
@@ -323,6 +323,110 @@ UniValue importaddress(const JSONRPCRequest& request)
     return NullUniValue;
 }
 
+static void RemoveAddress(CWallet*, const CTxDestination& dest, bool purgeTxs);
+static void RemoveScript(CWallet* const pwallet, const CScript& script, bool isRedeemScript, bool purgeTxs) EXCLUSIVE_LOCKS_REQUIRED(pwallet->cs_wallet)
+{
+    isminetype ismine = pwallet->IsMine(script);
+    if (!isRedeemScript && ismine == ISMINE_SPENDABLE) {
+        throw JSONRPCError(RPC_WALLET_ERROR, "The wallet contains the private key for this address");
+    } else if (!isRedeemScript && ismine == ISMINE_NO) {
+        throw JSONRPCError(RPC_WALLET_ERROR, "The wallet does not contain this address or script");
+    }
+    assert(ismine == ISMINE_WATCH_ONLY || isRedeemScript);
```

Comment:
This will be caught too as `CHECK_NONFATAL` is recommended instead of `assert` for rpc code.